### PR TITLE
Requesting a pull to mozilla:master from k88hudson:t1801

### PR DESCRIPTION
**Pull request by k88hudson**

### CODE REVIEW

File: templates/assets/plugins/titles/popcorn.titles.js [hunk 1]
```diff
@@ -32,110 +21,156 @@
...
+          values: [ "center", "bottom", "left", "right", "top", "custom"  ],
+          label: "Text Position",
+          "default": "center"
+        },
+        start: {
           elem: "input",
           type: "text",
-          label: "Out"
+          label: "In",
+          group: "advanced"
```
Comment (mjschranz) [inline templates/assets/plugins/titles/popcorn.titles.js:68]: Wait, why do you want start and end in the advanced options? Or did I just accidentally do that and never noticed?

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/380c6801dff8448/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/380c6801dff8448/output.txt

Total script time: 0.65 mins

Server is running on http://pr799.butyr.org/templates/basic/
Run the Test Suite: http://pr799.butyr.org/test/

File: templates/assets/editors/titles/titles-editor.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+            option;
+
+        function togglePopup() {
+          triangleObject.element.parentNode.style.display = "none";
+          flipObject.element.parentNode.style.display = "none";
+          soundObject.element.parentNode.style.display = "block";
+          iconObject.element.parentNode.style.display = "block";
+        }
+
+        function toggleSpeech() {
```
Comment (humphd) [inline templates/assets/editors/titles/titles-editor.js]: We don't seem to support this in the UI, should we remove?

File: templates/assets/editors/titles/titles-editor.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+              target.view.blink();
+            }
+            else {
+              _butter.currentMedia.view.blink();
+            }
+          }, false );
+        }
+
+        for ( key in pluginOptions ) {
+          if ( pluginOptions[ key ] ) {
```
Comment (humphd) [inline templates/assets/editors/titles/titles-editor.js]: Use `pluginOptions.hasOwnProperty( key )`

File: templates/assets/plugins/titles/popcorn.titles.js [hunk 4]
```diff
@@ -32,110 +21,157 @@
...
+        author: "@k88hudson, @mjschranz"
       },
       options: {
-        start: {
+        text: {
           elem: "input",
           type: "text",
-          label: "In"
+          label: "Text",
+          "default": "Popcorn.js"
```
Comment (humphd) [inline templates/assets/plugins/titles/popcorn.titles.js]: "Mozilla Popcorn" ?

File: templates/assets/plugins/titles/popcorn.titles.js [hunk 5]
```diff
@@ -32,110 +21,157 @@
...
+          units: "px",
+          group: "advanced"
         },
-        containerClasses: {
+        fontColor: {
           elem: "input",
           type: "text",
-          label: "Container Classes (separate with a space)",
-          "default": "popcorn-transition popcorn-titles"
+          label: "Font",
```
Comment (humphd) [inline templates/assets/plugins/titles/popcorn.titles.js]: "Font Colour"

File: templates/assets/plugins/titles/popcorn.titles.js [hunk 6]
```diff
@@ -32,110 +21,157 @@
...
+        transition: {
+          elem: "select",
+          options: [ "None", "Pop", "Fade", ],
+          values: [ "none", "pop", "popcorn-fade" ],
+          label: "Transition",
+          "default": "popcorn-fade"
         },
-        innerContainerClasses: {
+        fontFamily: {
+          elem: "select",
```
Comment (humphd) [inline templates/assets/plugins/titles/popcorn.titles.js:84]: Need some options to go with this select.

File: templates/assets/plugins/titles/popcorn.titles.js [hunk 7]
```diff
@@ -32,110 +21,157 @@
...
+          hidden: true
         }
       }
     },
 
     _setup: function( options ) {
 
-      var target,
+      var target = Popcorn.dom.find( options.target ),
           text = options.text,
```
Comment (humphd) [inline templates/assets/plugins/titles/popcorn.titles.js]: You should probably do some work to escape the text, or HTML will be an issue...

File: templates/assets/plugins/titles/popcorn.titles.js [hunk 8]
```diff
@@ -32,110 +21,157 @@
...
+      options._container.classList.add( "off" );
+
+      // Handle all custom fonts/styling
+      innerContainer.innerHTML = "<span>" + text + "</span>";
+      container.style.zIndex = +options.zindex;
       container.appendChild( innerContainer );
       target.appendChild( container );
 
+      container.style.fontStyle = options.fontItalics ? "italic" : "normal";
+      container.style.color = options.fontColor ? options.fontColor : "#668B8B";
```
Comment (humphd) [inline templates/assets/plugins/titles/popcorn.titles.js]: Should you validate this at all?

File: templates/assets/plugins/titles/popcorn.titles.js [hunk 9]
```diff
@@ -32,110 +21,157 @@
...
+      container.style.textDecoration = options.textUnderline ? "underline" : "none";
+      container.style.fontSize = options.fontSize ? options.fontSize + "px" : "25px";
```
Comment (humphd) [inline templates/assets/plugins/titles/popcorn.titles.js]: Do you have max/min sizes you allow?  A font size of -6 will be `true` here.

File: templates/basic/saved-data.json [hunk 10]
```diff
@@ -29,12 +29,7 @@
                 "start": 0.076,
                 "end": 1.085,
                 "target": "video-container",
-                "text": "Popcorn.js",
-                "innerContainerClasses": "",
-                "containerClasses": "popcorn-transition popcorn-titles",
-                "activeClass": "popcorn-on",
-                "inactiveClass": "popcorn-off",
-                "transitionSpeed": "normal"
+                "text": "Popcorn.js"
```
Comment (humphd) [inline templates/basic/saved-data.json]: "Mozilla Popcorn"

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/5961dcd68814734/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/5961dcd68814734/output.txt

Total script time: 1.11 mins

File: templates/assets/editors/titles/titles-editor.js (hunk 2 above)
Comment (k88hudson) [inline templates/assets/editors/titles/titles-editor.js]: Yes, not sure why this is here

File: templates/assets/plugins/titles/popcorn.titles.js (hunk 6 above)
Comment (k88hudson) [inline templates/assets/plugins/titles/popcorn.titles.js:84]: this is a googleFonts module, so setting  `googleFonts = true` adds a list of fonts

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/73c4cc9367b355f/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/45e838d87dfaea4/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/73c4cc9367b355f/output.txt

Total script time: 1.04 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/45e838d87dfaea4/output.txt

Total script time: 0.63 mins

Server is running on http://pr799.butyr.org/templates/basic/
Run the Test Suite: http://pr799.butyr.org/test/

**Pull request merged**